keanu-project/src/main/java/io/improbable/keanu/network/ProtobufSaver.java

```diff
@@ -147,6 +149,20 @@ public void saveValue(BoolVertex vertex) {
         }
     }
 
+    private KeanuSavedBayesNet.StoredValue getValue(Vertex vertex, String formattedValue) {
+        KeanuSavedBayesNet.GenericTensor savedValue = KeanuSavedBayesNet.GenericTensor.newBuilder()
+            .addAllShape(Arrays.asList(1l, 1l))
```

Comment:
Primarily because the only one that falls in here are generic vertices and at the moment we don't have a good story on how to save and load generic values outside of strings so there's only going to be one value - you are correct that throwing out the shape is probably silly though so I'll change to save the shape explicitly still.